src/graphics.c

```diff
@@ -3098,6 +3119,12 @@ struct line *line_new(void)
 	l->layout = NULL;
 	l->align = 0;
 	l->pos = 0;
+
+	#ifdef _USE_HARFBUZZ
+	/* Use the game language direction as inital value. */
+	l->direction = game_lookup(curgame_dir)->rtl? HB_DIRECTION_RTL: HB_DIRECTION_LTR;
```

Comment:
Could here game_lookup() return NULL? I think yes, while we render menu and no game is selected/available.
The next thing i am not sure, is that currently line and layout are abstracted from games. And this code break this abstraction.

How about adding direction to layout? 
Then after line_new() call we will inherit direction from layout (line->direction = layout->direction) 
Add txt_layout_direction() function to set direction for layout.
Make txt_layout_direction() call from game.c in game_apply_theme()

What do you think?

Reply:
Yes, it can return null and the direction would be left-to-right. btw, direction is only used in RTL logic. I have not yet fixed the menus completely, but I have some ideas. First we should agree on the basic RTL functionality.

> The next thing i am not sure, is that currently line and layout are abstracted from games. 

My question here is, what is a line direction? For me, direction depends on the text. Arabic is right to left, Russian is left to right. So it is natural to use the text for direction. 

> How about adding direction to layout?

I think layout can have its own direction too, but it is independent from the text direction. For example, if you go to developer tools in Android, there is a Force RTL Layout option. If you enable it, every UI element will be displayed from rith to left, but it does not change the text and line directions. It is always derived from the content.

> Then after line_new() call we will inherit direction from layout 

This will only works if a text has only one direction, which is not true. See my last picture in the #76 . The Latin text is left-to-right, the Arabic text is right-to-left. We can not derive this from layout.

Reply:
> Yes, it can return null and the direction would be left-to-right. btw, direction is only used in RTL logic.

I mean that this code will crash while the game is not selected, game_lookip(curgame_dir) == NULL in that case.

>  For me, direction depends on the tex. ... I think layout can have its own direction too, but it is independent from the text direction.

Ok, i see, but in you code direction is set on every line_new, taken from game structure. So, i think it is default direction? Is the game.rtl flag is sitll needed? What game.rtl flag used for?

Reply:
> I mean that this code will crash while the game is not selected, game_lookip(curgame_dir) == NULL

Now I undestand. Thanks, I'll fix this.

> direction is set on every line_new, taken from game structure. So, i think it is default direction?

I see. Yes, that is only the default direction. Just like how thing works now. The real direction comes from checking the words.

Reply:
So, if it is default direction, i just think to move it to layout. Like other properties.
line_new is the internal function for graphics.c and it is good to avoid to do game_XXX calls from graphics.c

game_apply_theme constructs layouts for game output, so, here we can put default direction line txt_layout_direction (or default_direction).

Then, after line_new move this property to line. Then, if needed, to word...

Reply:
I see. I'll move the game_lookup call to layout and just hand it over...

Reply:
I just made those changes...